fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -594,14 +596,77 @@ public Set<String> getUpdatedPartitionNamesOfOlapTable(OlapTable baseTable) {
         return result;
     }
 
-    private boolean isLessThanMVRewriteStaleness() {
+    /**
+     * @return Return max timestamp of all table's max refresh timestamp
+     *          which is computed by checking all its partitions' modified time.
+     */
+    public Optional<Long> maxBaseTableRefreshTimestamp() {
+        long maxRefreshTimestamp = -1;
+        for (BaseTableInfo baseTableInfo : baseTableInfos) {
+            Table baseTable = baseTableInfo.getTable();
+            if (baseTable instanceof OlapTable) {
```

Comment:
we have many if-else branch for the `BaseTableInfo`. 

we should refactor the `BaseTableInfo` into the Inheritance style, with `OlapBaseTableInfo extends BaseTableInfo`, `HiveBaseTableInfo extends BaseTableInfo`

Reply:
Before we can only support olapTable and HiveTable's base table infos.
Later we may support more and more Tables as BaseTable, it may be a big change, Maybe we can do it later.